### PR TITLE
Add support for user banner and bannerAccentColor

### LINKED ISSUE #965: Support user banners
**Feature Description:** 
Support user banners.

**Justification:** 
https://github.com/discord/discord-api-docs/pull/3448

### PR DESCRIPTION
**Pull request by Azn9**
**Description:** Add methods User#getBannerUrl(ImageFormat) and User#getBannerAccentColor()

**Justification:** 
Closes #965 
Require https://github.com/Discord4J/discord-json/pull/103

### CODE REVIEW

**Comment by quanticc** [comment]
on hold until discord docs issue is merged

File: core/src/main/java/discord4j/core/object/entity/User.java [hunk 1]
```diff
@@ -129,6 +132,26 @@ public final boolean hasAnimatedAvatar() {
...
 
+    /**
+     * Gets the user's banner URL, if present.
+     *
+     * @param format The format for the URL.
+     * @return The user's banner URL, if present.
+     */
+    public final Optional<String> getBannerUrl(final Image.Format format) {
+        return data.banner().toOptional().orElse(Optional.empty()).map(avatar -> ImageUtil.getUrl(
+                String.format(BANNER_IMAGE_PATH, getId().asString(), avatar), format));
```
Comment (darichey) [inline core/src/main/java/discord4j/core/object/entity/User.java:143]:
```suggestion
        return Possible.flatOpt(data.banner()).map(avatar -> ImageUtil.getUrl(
                String.format(BANNER_IMAGE_PATH, getId().asString(), avatar), format));
```

File: core/src/main/java/discord4j/core/object/entity/User.java [hunk 2]
```diff
@@ -129,6 +132,26 @@ public final boolean hasAnimatedAvatar() {
...
+    }
+
+    /**
+     * Gets the user's banner accent color, if present.
+     *
+     * @return The user's banner accent color, if present.
+     */
+    public final Optional<Integer> getAccentColor() {
+        return data.accentColor().toOptional().orElse(Optional.empty());
+    }
```
Comment (darichey) [inline core/src/main/java/discord4j/core/object/entity/User.java:153]:
```suggestion
    public final Optional<Color> getAccentColor() {
        return Possible.flatOpt(data.accentColor())
                .map(Color::of);
    }
```

**Review by darichey — CHANGES_REQUESTED** [review]
Great start! Just a couple of things:
* We can use `Possible.flatOpt()` to flatten the `Possible<Optional<T>>` to `Optional<T>` more nicely
* Let's convert the integer to a `Color` for convenience and consistency 
* Let's also add a `getBanner(Image.Format)` that can look exactly like `getAvatar`, again for convenience and consistency

**Review by darichey — APPROVED** [review]
Looks great, thanks!

**Pull request merged**